CMakeLists.txt

```diff
@@ -12,53 +10,61 @@ target_include_directories(clap-core INTERFACE include)
 
 install(DIRECTORY include DESTINATION "." OPTIONAL EXCLUDE_FROM_ALL)
 
-add_executable(clap-compile-c EXCLUDE_FROM_ALL src/main.c)
+option(CLAP_BUILD_TESTS "Should CLap build tests and the like?" OFF)
 
-macro(clap_compile_cpp SUFFIX EXT STDC STDCPP)
-    add_executable(clap-compile-${SUFFIX} EXCLUDE_FROM_ALL src/main.${EXT})
-    target_link_libraries(clap-compile-${SUFFIX} clap-core)
-    set_target_properties(clap-compile-${SUFFIX} PROPERTIES
-        C_STANDARD ${STDC}
-        CXX_STANDARD ${STDCPP})
-    add_test(NAME test-clap-compile-${SUFFIX} COMMAND clap-compile-${SUFFIX})
-    add_dependencies(clap-tests clap-compile-${SUFFIX})
+if (${CLAP_BUILD_TESTS})
+    message(STATUS "Including clap tests, compile tests, and versions")
+    add_custom_target(clap-tests)
 
-    if (${CMAKE_C_COMPILER_ID} STREQUAL "GNU" OR ${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
-        target_compile_options(clap-compile-${SUFFIX} PRIVATE -Wall -Wextra -pedantic)
-    endif()
+    add_executable(clap-compile-c EXCLUDE_FROM_ALL src/main.c)
 
-    if (${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
-        target_compile_options(clap-compile-${SUFFIX} PRIVATE -Werror=pragma-pack)
-    endif()
-endmacro()
+    macro(clap_compile_cpp SUFFIX EXT STDC STDCPP)
```

Comment:
Could we detect if a C or C++ standard is supported here?

Reply:
Let's not do detection it is painful and not worth it.
All we need is the github action to cover the tests.